icepyx/core/read.py

```diff
@@ -752,9 +817,16 @@ def load(self):
         # In these situations, xarray recommends manually controlling the merge/concat process yourself.
         # While unlikely to be a broad issue, I've heard of multiple matching timestamps causing issues for combining multiple IS2 datasets.
         for file in self.filelist:
+            if file.startswith("s3"):
+                # If path is an s3 path create an s3fs filesystem to reference the file
+                # TODO would it be better to be able to generate an s3fs session from the Mixin?
+                s3 = earthaccess.get_s3fs_session(daac="NSIDC")
+                file = s3.open(file, "rb")
+
             all_dss.append(
                 self._build_single_file_dataset(file, groups_list)
             )  # wanted_groups, vgrp.keys()))
+            file.close()
```

Comment:
this line causes an error when `file` is a string pointing to a local file (`AttributeError: 'str' object has no attribute 'close'`).

Reply:
Good catch. Just pushed a fix.